### PR TITLE
[Task]: clear the open Dependabot alerts — 12 stale-manifest ghosts, 3 fixable floors, 3 blocked upstream

### LINKED ISSUE #1036: [Task]: clear the open Dependabot alerts — 12 stale-manifest ghosts, 3 fixable floors, 3 blocked upstream
### Description

Triage and clear the 20 open Dependabot alerts. 12 are ghosts against manifests deleted from `develop` in June; 2 are fixable with dependency floors; 1 needs a Cypress major (tracked separately); 5 are blocked upstream and stay open deliberately.

### Background

The board has accumulated because the repo has `dependabot_security_updates: disabled` and no `.github/dependabot.yml`, so no alert ever gets a PR attached. The Security tab also reports *"Dependency file checks have 1 error."*

**The 12 ghosts** name manifests that no longer exist on `develop` or `main`:

| Manifest | Alerts | Removed by |
|---|---|---|
| `fl_services/fl-api-base/uv.lock` | #503 #504 #505 (aiohttp), #506 #507 #508 (cryptography) | renamed in `fd36c0a9` + `7676a4e3` → `fl-services/nvflare/fl-api-base/uv.lock` |
| `fl-apps/tutorials/image_segmentation/3d_spleen_segmentation/uv.lock` | #480 #516 #517 #518 (monai), #473 (setuptools), #440 (torch) | deleted in `fecc1f4a`; tutorials moved to `fl-tutorials/` and deliberately do not commit lockfiles (`.gitignore:463`) |

GitHub's dependency-graph snapshot still carries both, so every newly-published advisory keeps minting alerts against dead files. The live successors are already patched — `cryptography 50.0.0`, `aiohttp 3.14.3`, `monai 1.6.0` — via the `[tool.uv] constraint-dependencies` blocks the repo already maintains for exactly this purpose. #440 and #473 are ghost duplicates of live #439 and #475.

**The 2 fixable:**

- **#515 nanoid** (High, CVE-2026-67213) — 3.3.16, patched in 3.3.18. Reaches production scope via `vue → @vue/compiler-sfc → postcss → nanoid`.
- **#512 js-yaml** (High, GHSA-5p4m-2wfm-xmqj) — 3.15.0, patched in 3.15.1. Already floored by an existing `overrides` entry at `>=3.15.0 <4.0.0`; one character. The `<4.0.0` bound must stay: `@istanbuljs/load-nyc-config` peer-requires the v3 line, and 4.x is also affected and unpatched.

**Deliberately left open:**

- **#439 torch** and **#475 setuptools** — both gated on the same pin. No torch >=2.13.0 build runs on the dev fleet (cu126 has no `sm_120`, cu128 stops at 2.11.0, cu130 needs driver >=580), and **`torch==2.11.0+cu128` itself declares `setuptools<82`**, so a `setuptools>=83.0.0` constraint is unresolvable — `uv lock` refuses it outright. torch 2.13.0 drops that ceiling, so both alerts close together with the cu130 migration decision in #826. Not force-overridden: torch's ceiling exists because setuptools 82 removed `pkg_resources`, and the setuptools advisory is a `MANIFEST.in` sdist-exclusion bypass on macOS APFS/HFS+ — this repo has no `MANIFEST.in`, no setuptools build backend outside `flip-api` (which ships as an image, never an sdist), and no macOS runner.
- **#258 / #400 flask** (Low x2) — `nvflare==2.8.0` and 2.8.1 both hard-pin `Flask==3.0.2`, so no constraint can lift it. FLIP imports no Flask and never launches nvflare's `dashboard`/`edge` modules, its only Flask consumers. **nvflare 2.9.0rc1 pins `Flask==3.1.3`**, so the fix lands with the nvflare 2.9.0 upgrade — #1037.

**#514 extract-zip** (High, dev) has no patched release at all; Cypress dropped the dependency in 15.16.0. Tracked separately as a Cypress 14 → 15 upgrade.

### Acceptance criteria

- [x] `flip-ui/package.json` overrides floor `js-yaml` at `>=3.15.1 <4.0.0` and `nanoid` at `>=3.3.18 <4.0.0`; `npm ls nanoid js-yaml` shows only patched versions and `npm audit` reports nothing but `extract-zip`.
- [x] Each alert that stays open is documented at the point of omission, so the next reader finds the reason without re-deriving it: the `# flask>=3.1.3 omitted:` blocks in `flip-utils/pyproject.toml` and `fl-services/nvflare/fl-api-base/pyproject.toml` name #1037, and a matching `# setuptools>=83.0.0 omitted:` block names #826.
- [x] No `uv.lock` is modified — the Python-side changes are comments only, and `uv lock --check` passes for both projects.
- [x] The 12 ghost alerts are closed — either by a dependency-graph re-scan dropping the dead manifests, or dismissed as `not_used` with the rename/deletion commit cited.
- [ ] `gh api /repos/londonaicentre/FLIP/dependabot/alerts?state=open` returns five rows: #258, #400 (flask), #439, #475 (torch pin), #514 (Cypress, tracked separately).

### Prerequisites

- `uv` 0.11.16 locally (an older uv rewrites `exclude-newer` in every lockfile).
- For the dismissals: `security_events` write scope on the `gh` token, and admin or security-manager on the repo.

### Known or potential blockers

Fixing the *"Dependency file checks have 1 error"* is the cleaner route for the 12 ghosts than dismissing them, because it also stops the next advisory from minting a 13th — but it may not be resolvable from the repo side, in which case dismissal is the fallback.

- [x] I confirm I have assigned the relevant label(s) and, if known, Milestone to this task

### PR DESCRIPTION
**Pull request by atriaybagur**
# Description

Closes the two Dependabot alerts that are actually fixable, and records at the point of omission why each remaining one is not — so the next reader finds the reason in the file rather than re-deriving it.

## What changed

**`flip-ui/package.json` overrides** — two floors, both verified resolved:

| Package | Was | Now | Alert |
|---|---|---|---|
| `js-yaml` | `>=3.15.0 <4.0.0` | `>=3.15.1 <4.0.0` | #512, GHSA-5p4m-2wfm-xmqj (High) |
| `nanoid` | *(none)* | `>=3.3.18 <4.0.0` | #515, CVE-2026-67213 (High) |

- The `js-yaml` `<4.0.0` bound stays deliberately: `@istanbuljs/load-nyc-config` peer-requires the v3 line, and 4.x is *also* affected with the fix not backported. 3.15.1 is the `v3-legacy` dist-tag.
- `nanoid` carries production scope because it reaches the graph through `vue → @vue/compiler-sfc → postcss → nanoid`, not only the dev chain. `postcss` asks for `^3.3.16`, so the 3.x `legacy` line satisfies it and the `<4.0.0` bound keeps it there (`latest` is 6.0.1). An override rather than a bare `npm update` matches the 14 existing entries in that block and survives re-resolution.

The `package-lock.json` diff is exactly those two packages. `js-yaml` also hoists from `@istanbuljs/load-nyc-config/node_modules/` to top level (with its `argparse`), which is just npm de-duplicating now that only one copy exists.

**Two comment blocks** in `flip-utils/pyproject.toml` and `fl-services/nvflare/fl-api-base/pyproject.toml`, documenting the omissions. No Python lockfile is regenerated — these edits are comments only, and `uv lock --check` passes unchanged for both projects.

## Why setuptools (#475) is a comment and not a floor

This started as a planned one-line `setuptools>=83.0.0` constraint. `uv lock` rejects it:

```
Because torch==2.11.0+cu128 depends on setuptools<82 and setuptools>=83.0.0,
we can conclude that torch==2.11.0+cu128 cannot be used.
```

Verified against PyPI: torch 2.11.0 declares `setuptools<82`, torch 2.13.0 drops the ceiling, and 81.0.0 is the only 81.x release. The torch pin is itself fixed by the cu128 index, because no torch >=2.13.0 build runs on the dev fleet (cu126 has no `sm_120`, cu128 stops at 2.11.0, cu130 needs driver >=580).

So **#475 is gated on the same decision as #439** and closes with it under #826 — it is not an independent fix.

Not force-overridden. torch caps setuptools because 82 removed `pkg_resources`, so an override risks `torch.utils.cpp_extension` at runtime in the FL images; and the advisory is a `MANIFEST.in` sdist-exclusion bypass on macOS APFS/HFS+, while this repo has no `MANIFEST.in` anywhere, no setuptools build backend outside `flip-api` (which ships as an image, never an sdist), and no macOS runner. Bad trade.

## Alert board after this PR

The other 12 open alerts are **ghosts** — they name manifests deleted from `develop` in June (`fl_services/fl-api-base/uv.lock`, renamed in `fd36c0a9`+`7676a4e3`; `fl-apps/tutorials/.../3d_spleen_segmentation/uv.lock`, deleted in `fecc1f4a`). The live successors are already patched (`cryptography 50.0.0`, `aiohttp 3.14.3`, `monai 1.6.0`). They need a dependency-graph refresh or dismissal, not a code change — handled outside this PR under #1036.

What legitimately stays open, each now documented in-tree: #439 + #475 (torch pin, #826), #258 + #400 (flask, blocked until nvflare 2.9.0 — #1037), #514 (extract-zip, no patched release exists; needs the Cypress 14 → 15 upgrade, tracked separately).

## Linked Issues

Fixes #1036

The ghost-alert dismissals and the follow-up issues that #1036 also called for are already done outside this PR: alerts #440, #473, #480, #503-#508 and #516-#518 are dismissed as `not_used`, and #1037 (nvflare 2.9.0), #1039 (Cypress 15), #1040 (map-apps monai ceiling) and #1041 (flip-ui dev-scope drift) are filed. Merging this closes the last of it.

## Checklist

- [x] Follows the project's coding conventions and style guide
- [x] Updates documentation — the rationale lives in the pyproject comments, which is where the existing omission notes live
- [ ] I have added tests that prove my fix is effective or that my feature works — n/a, dependency floors; `npm ls` / `npm audit` are the check
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published

## Type of Change

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make -C docs/ docs`.

## Testing

```
$ npm ls nanoid js-yaml
flip-ui@0.5.0
├─┬ postcss@8.5.25
│ └── nanoid@3.3.18
└─┬ ts-jest@29.4.6
  └─┬ @jest/transform@30.3.0
    └─┬ babel-plugin-istanbul@7.0.1
      └─┬ @istanbuljs/load-nyc-config@1.1.0
        └── js-yaml@3.15.1

$ npm audit
extract-zip  *   High   (only remaining finding; no patched release, needs Cypress 15)

$ npm run lint          # clean
$ npm run test:unit     # Test Files 112 passed (112) | Tests 1266 passed, 1 skipped (1267)

$ uv lock --check --project flip-utils                        # Resolved 148 packages
$ uv lock --check --project fl-services/nvflare/fl-api-base   # Resolved 93 packages
$ git status --porcelain -- '*uv.lock'                        # empty — no Python lock touched
```

- [ ] Quick tests passed locally by running `make unit-test` — ran the flip-ui suite directly; no Python source changed, so the Python suites are unaffected.
- [ ] Integration tests pass (if applicable) by running `make integration-test` — n/a.

## Additional Notes

`uv` 0.11.16 was used throughout, matching the pins in `.pre-commit-config.yaml` and the `dependency-cooldown` CI gate, so no lockfile picked up a rewritten `exclude-newer`.

## Acceptance Criteria

*Imported from issue #1036*

- [x] `flip-ui/package.json` overrides floor `js-yaml` at `>=3.15.1 <4.0.0` and `nanoid` at `>=3.3.18 <4.0.0`; `npm ls nanoid js-yaml` shows only patched versions and `npm audit` reports nothing but `extract-zip`.
- [x] Each alert that stays open is documented at the point of omission, so the next reader finds the reason without re-deriving it: the `# flask>=3.1.3 omitted:` blocks in `flip-utils/pyproject.toml` and `fl-services/nvflare/fl-api-base/pyproject.toml` name #1037, and a matching `# setuptools>=83.0.0 omitted:` block names #826.
- [x] No `uv.lock` is modified — the Python-side changes are comments only, and `uv lock --check` passes for both projects.
- [x] The 12 ghost alerts are closed — either by a dependency-graph re-scan dropping the dead manifests, or dismissed as `not_used` with the rename/deletion commit cited.
- [ ] `gh api /repos/londonaicentre/FLIP/dependabot/alerts?state=open` returns five rows: #258, #400 (flask), #439, #475 (torch pin), #514 (Cypress, tracked separately).

  *(5 is the only one that cannot be ticked pre-merge: the board currently shows seven rows — those five plus #512 and #515, which close when this merges. Verified today that the other twelve are gone.)*

### CODE REVIEW

**Comment by atriaybagur** [comment]
Ready for review.

**CI:** 33 green, 1 skipped — the skip is *Ensure PR to main originates from develop*, which is the expected outcome for a develop-targeted PR. `MERGEABLE`, and level with `develop` (0 commits behind).

**Tests:** `npm run lint` clean; `npm run test:unit` → 112 files, 1266 passed, 1 skipped. No Python source changed, so the Python suites are unaffected — and no `uv.lock` is touched at all, since the Python-side edits here are comments only. `uv lock --check` passes unchanged for both `flip-utils` and `fl-services/nvflare/fl-api-base`.

**What to look at:** the two npm floors are mechanical (`js-yaml` `>=3.15.0` → `>=3.15.1`, new `nanoid` `>=3.3.18 <4.0.0`), and the `package-lock.json` diff is exactly those two packages — `js-yaml` also hoists out of `@istanbuljs/load-nyc-config/node_modules/` because only one copy remains.

The part worth a reviewer's eye is the **setuptools decision**, which changed during the work. It was planned as a one-line `setuptools>=83.0.0` constraint; `uv lock` rejects it, because `torch==2.11.0+cu128` declares `setuptools<82` and the torch pin is itself fixed by the cu128 index. Confirmed upstream: torch 2.11.0 caps it, torch 2.13.0 drops the cap, and 81.0.0 is the only 81.x release. So Dependabot #475 is gated on the same decision as #439 and closes with it under #826 — it is not an independent fix. I left a comment at the point of omission rather than forcing an override past torch's ceiling, since setuptools 82 removed `pkg_resources` and the advisory itself is a `MANIFEST.in` sdist bypass on macOS APFS/HFS+ — against a repo with no `MANIFEST.in`, no setuptools build backend outside `flip-api` (which ships as an image, never an sdist), and no macOS runner.

**Board impact:** the 12 stale-manifest alerts are already dismissed as `not_used` (manifests deleted from `develop` in June; the live successors are on `cryptography 50.0.0` / `aiohttp 3.14.3` / `monai 1.6.0`). Merging this closes #512 and #515, leaving four deliberate: #439 + #475 (torch pin, #826), #258 + #400 (flask, blocked until nvflare 2.9.0 — #1037), plus #514 which #1042 handles.

**Copilot:** review requested.

**Review by garciadias — APPROVED** [review]
Approving. Every checkable claim in the PR body verifies against live Dependabot alert state, the actual lockfiles, and PyPI advisory/version metadata: the js-yaml/nanoid floor bumps land exactly on their advisories' first_patched_version, the flask/nvflare and setuptools/torch pin comments accurately describe genuinely unresolvable upstream constraints, and a sample of the 12 dismissed "ghost" alerts checks out against the rename commits and current lockfile successors. CI is fully green including the dependency cooldown gate.

One non-blocking note: the monai "ghost" dismissal (#480/#516-518) says "no successor manifest exists on develop," but fl-tutorials/nvflare/image_segmentation/3d_spleen_segmentation/pyproject.toml does float monai with no version ceiling and no exclude-newer block, unlike its Flower sibling which correctly floors >=1.6.0. It's not exploitable today (no reachable vulnerable sink in that tutorial, and monai's current latest happens to already be patched), but worth a quick follow-up to floor it explicitly and add the same cooldown block as the Flower tutorial, so it doesn't just get lucky on the next monai release.
